include/rmm/mr/device/logging_resource_adaptor.hpp

```diff
@@ -59,15 +70,29 @@ class logging_resource_adaptor final : public device_memory_resource {
    * @param filename Name of file to write log info. If not specified, retrieves
    * the file name from the environment variable "RMM_LOG_FILE".
    */
-  logging_resource_adaptor(Upstream* upstream, std::string const& filename =
-                                                   get_default_filename())
-      : upstream_{upstream},
-        logger_{std::make_shared<spdlog::logger>(
-            "RMM", std::make_shared<spdlog::sinks::basic_file_sink_mt>(
-                       filename, true /*truncate file*/))} {
+  logging_resource_adaptor(Upstream* upstream,
+                           LogLocation location = LogLocation::FILE,
```

Comment:
Instead of an enum switch, just add a new constructor that takes osstream and pass that to a `ostream_sink` from spdlog.

Reply:
Done.